pandas/io/sas/sas_xport.py

```diff
@@ -267,7 +267,11 @@ def __init__(
             # should already be opened in binary mode in Python 3.
             self.filepath_or_buffer = filepath_or_buffer
 
-        self._read_header()
+        try:
+            self._read_header()
+        except:  # noqa: E722  OK because exception re-raised
```

Comment:
*never* use bare exceptions, use
```
try:
   ....
except Exception:
   # close on error reading header...
   ....
```

Reply:
I can do that, but what about errors that don't inherit from Exception?
I note that the PEP says:

> A good rule of thumb is to limit use of bare 'except' clauses to two cases:
>    If the exception handler will be printing out or logging the traceback; at least the user will be aware that an error has occurred.
 >   If the code needs to do some cleanup work, but then lets the exception propagate upwards with raise. try...finally can be a better way to handle this case.

(try...finally obviously isn't applicable here since we only want to close on error)

Reply:
simply use Exception

Reply:
OK, I note that exceptions that don't inherit from Exception are supposed to be system-exiting, so we can suppose some higher-level process will clean up for us in this case.